site/src/pages/rules.md

```diff
@@ -388,3 +388,13 @@ The use of fallback array feature is not recommended. It currently does not have
 Related issues: [nodejs/node#37928](https://github.com/nodejs/node/issues/37928), [vitejs/vite#4439](https://github.com/vitejs/vite/issues/4439), [microsoft/TypeScript#50762](https://github.com/microsoft/TypeScript/issues/50762), [webpack/enhanced-resolve#400](https://github.com/webpack/enhanced-resolve/issues/400), [evanw/esbuild#2974](https://github.com/evanw/esbuild/issues/2974), [web-infra-dev/rspack#5052](https://github.com/web-infra-dev/rspack/issues/5052)
 
 (Works similarly to [EXPORTS_FALLBACK_ARRAY_USE](#exports_fallback_array_use)).
+
+## `CJS_WITH_ESMODULE_DEFAULT_EXPORT` {#cjs_with_esmodule_default_export}
+
+It is discouraged to use `default` exports in CJS files that are transpiled from ESM as the default export may have a different value depending on the bundler or the runtime.
+
+CJS files that are transpiled from ESM have a `__esModule` marker property, which are used by bundlers to determine if the file was originally written in ESM. However, Node.js does not treat files with `__esModule` specially, and does not allow using named exports from those files.
```

Comment:
One last thing, is this part accurate?

> and does not allow using named exports from those files

Doesn't node use `cjs-module-lexer` to get the named exports so it works too? Regardless of `__esModule` since it's used by bundlers to identify the default export, named exports will always work between CJS-ESM.